units/src/amount/unsigned.rs

```diff
@@ -423,57 +423,93 @@ impl fmt::Display for Amount {
 }
 
 impl ops::Add for Amount {
-    type Output = Amount;
+    type Output = AmountOpResult<Amount>;
 
     fn add(self, rhs: Amount) -> Self::Output {
-        self.checked_add(rhs).expect("Amount addition error")
+        match self.checked_add(rhs) {
+            Some(amount) => AmountOpResult::Valid(amount),
+            None => AmountOpResult::Error(AmountOpError {}),
+        }
     }
 }
-crate::internal_macros::impl_add_for_references!(Amount);
-crate::internal_macros::impl_add_assign!(Amount);
+impl ops::Add<Amount> for &Amount {
+    type Output = AmountOpResult<Amount>;
+
+    fn add(self, rhs: Amount) -> Self::Output { *self + rhs }
+}
+
+impl ops::Add<&Amount> for Amount {
+    type Output = AmountOpResult<Amount>;
+
+    fn add(self, rhs: &Amount) -> Self::Output { self + *rhs }
+}
+
+impl<'a> ops::Add<&'a Amount> for &Amount {
+    type Output = AmountOpResult<Amount>;
+
+    fn add(self, rhs: &'a Amount) -> Self::Output { *self + *rhs }
+}
```

Comment:
Ops are not implemented for references in `core`. I wonder if there's some good reason to do that.

Reply:
> Ops are not implemented for references in core. 

I don't understand what you mean? References are implemented for `u32`? https://doc.rust-lang.org/std/primitive.u32.html#impl-Add%3Cu32%3E-for-%26u32